### DISCUSSION TITLE
oracel cdc 执行由全量数据同步结束转增量同步时报下述异常

### DISCUSSION THREAD

**Discussion by MonsterChenzhuo** [apache/flink-cdc#1183] [Q&A] [answered]
cdc版本：2.2.1
flink版本：1.3.6
oracle版本：11.2.0.2
报错内容：Caused by: Error : 1403, Position : 0, Sql = BEGIN DBMS_LOGMNR_D.BUILD (options => DBMS_LOGMNR_D.STORE_IN_REDO_LOGS); END;, OriginalSql = BEGIN DBMS_LOGMNR_D.BUILD (options => DBMS_LOGMNR_D.STORE_IN_REDO_LOGS); END;, Error Msg = ORA-01403: 未找到任何数据
ORA-06512: 在 "SYS.DBMS_LOGMNR_INTERNAL", line 5972
ORA-06512: 在 "SYS.DBMS_LOGMNR_INTERNAL", line 6070
ORA-06512: 在 "SYS.DBMS_LOGMNR_D", line 12
ORA-06512: 在 line 1

代码：

public class OracleToKafka{

    public static void main(String[] args) throws Exception {
        Properties properties = new Properties();
        properties.setProperty("event.processing.failure.handling.mode", "skip");
        properties.setProperty("database.tablename.case.insensitive", "false");
        properties.setProperty("debezium.log.mining.strategy", "online_catalog");
        properties.setProperty("debezium.log.mining.continuous.mine", "true");

        SourceFunction<String> sourceFunction = OracleSource.<String>builder()
                .hostname("")
                .port(1521)
                .database("")
                .schemaList("")
                .tableList("")
                .username("")
                .password("")
                .debeziumProperties(properties)
                .deserializer(new JsonDebeziumDeserializationSchema())
                .build();
        Configuration config = new Configuration();
        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment(config);
        DataStreamSource<String> dataStreamSource = env.addSource(sourceFunction).setParallelism(1);
        dataStreamSource.setParallelism(1).print("====>");

        env.execute("FlinkCDCOracle");
    }
}

**Comment by fuyun2024** [comment]
This is an error in the process of build data dictionary by debezium, can you execute this SQL : `BEGIN DBMS_LOGMNR_D.BUILD (options => DBMS_LOGMNR_D.STORE_IN_REDO_LOGS); END;`

**Reply by MonsterChenzhuo** [reply]
@fuyun2024 执行过一样报错，字典不写redo，换成写文件可以

**Reply by MonsterChenzhuo** [reply]
EXECUTE dbms_logmnr_d.build('dictionary.ora','/u01/zrprod/temp',options => dbms_logmnr_d.store_in_flat_file) 这种方式可以；
生成的字典文件1.8G，约需要1分多钟；
而在线redo存放字典直接报错
SYS@ZRPROD> BEGIN DBMS_LOGMNR_D.BUILD (options => DBMS_LOGMNR_D.STORE_IN_REDO_LOGS); END;
  2  /
BEGIN DBMS_LOGMNR_D.BUILD (options => DBMS_LOGMNR_D.STORE_IN_REDO_LOGS); END;
*
ERROR at line 1:
ORA-01403: no data found
ORA-06512: at "SYS.DBMS_LOGMNR_INTERNAL", line 5972
ORA-06512: at "SYS.DBMS_LOGMNR_INTERNAL", line 6070
ORA-06512: at "SYS.DBMS_LOGMNR_D", line 12
ORA-06512: at line 1

**Reply by fuyun2024** [reply]
Have you started archiving logs？

**Reply by MonsterChenzhuo** [reply]
@fuyun2024  您好，给您添麻烦了，是db的问题logmnr配置的有问题

**Comment by shadon178** [comment, marked as answer]
应该是oracle的问题，我试过‘**在线字典文件**’和‘**存储在归档日志中的字典文件**’都可以的。
